packages/pcd/gpc-pcd/src/validatorChecks.ts

```diff
@@ -0,0 +1,158 @@
+import { GPCProofConfig, PODMembershipLists } from "@pcd/gpc";
+import {
+  PODEntries,
+  PODName,
+  applyOrMap,
+  decodePublicKey,
+  podValueHash
+} from "@pcd/pod";
+import { PODPCD, PODPCDTypeName } from "@pcd/pod-pcd";
+import _ from "lodash";
+import {
+  PODEntryRecord,
+  PODPCDArgValidatorParams,
+  PODSignerPublicKeys
+} from "./GPCPCD";
+
+export function checkPCDType(podPCD: PODPCD): boolean {
+  return podPCD.type === PODPCDTypeName;
+}
+
+export function checkPODEntriesAgainstProofConfig(
+  podName: PODName,
+  podPCD: PODPCD,
+  proofConfig: GPCProofConfig | undefined,
+  params: PODPCDArgValidatorParams
+): proofConfig is GPCProofConfig {
+  // Check POD against serialised proof config if it is given.
+  if (proofConfig !== undefined) {
+    // POD podName should be present in the config and have all
+    // entries specified there.
+    const podConfig = proofConfig.pods[podName];
+    if (podConfig === undefined) {
+      params.notFoundMessage = `The proof configuration does not contain this POD.`;
+      return false;
+    }
+    const configEntries = Object.keys(podConfig.entries);
+
+    // Enumerate POD entries
+    const podEntries = podPCD.pod.content.asEntries();
+
+    // Return false if some entry in the config is not in the POD
+    if (
+      configEntries.some((entryName) => podEntries[entryName] === undefined)
+    ) {
+      return false;
+    } else {
+      return true;
+    }
+  } else {
+    params.notFoundMessage =
+      "The proof configuration has not been passed to the validator!";
+    return false;
```

Comment:
I think this should be true given the proofConfig is defined to be an optional argument.  I think requestors should have the choice to provide no validator parameters at all if they don't want to deal with it.  Alternately this question would go away if the validator had direct access to the args.